lib/msf/core/payload/php.rb

```diff
@@ -134,7 +134,8 @@ def php_system_block(options = {})
       }
     "
 
-    exec_methods = [passthru, shell_exec, system, exec, proc_open, popen].sort_by { rand }
+    exec_methods = [passthru, shell_exec, system, exec, proc_open, popen];
+    shuffle(exec_methods);
```

Comment:
Hey @jvoisin, there's an issue with this

Here is the fix: 

```suggestion
    exec_methods = %w(passthru shell_exec system exec proc_open popen)
    exec_methods.shuffle!
```

I was working on a new module and had this error:

`Exploit failed: NoMethodError undefined method shuffle`

Reply:
https://github.com/rapid7/metasploit-framework/pull/19491 !